v-next/hardhat-node-test-reporter/src/reporter.ts

```diff
@@ -244,45 +244,15 @@ export function hardhatTestReporter(
           yield event.data.message;
           break;
         }
-        case "test:plan": {
-          // Do nothing
-          break;
-        }
-        case "test:enqueue": {
-          // Do nothing
-          break;
-        }
-        case "test:dequeue": {
-          // Do nothing
-          break;
-        }
-        case "test:watch:drained": {
-          // Do nothing
-          break;
-        }
-        case "test:complete": {
-          // Do nothing
-          break;
-        }
         case "test:coverage": {
           yield chalk.red("\nTest coverage not supported by this reporter\n");
           break;
         }
-        // @ts-expect-error -- A new event type was added in https://github.com/nodejs/node/pull/54851
-        // but @types/node doesn't know about it yet. Once it does, the dependency should be updated
-        // and this comment removed.
-        case "test:summary": {
-          // Do nothing
-          break;
-        }
-        /* eslint-disable-next-line @typescript-eslint/switch-exhaustiveness-check --
-        We have this extra check here becase we know the @types/node type is
-        unreliable */
         default: {
```

Comment:
This looks good.
I was tempted to say we should keep the `do nothing` tests so we can explicitly see what events we are skipping (and consider whether we should add the actions for them).
But I think the complexity cost of keeping `@ts-expect-error` annoys me more than a theoretical future readability payoff.
What do you think?

Reply:
That was my thinking exactly. Also, when we want to reconsider what events we use, we should look at node directly anyway since @types/node, sadly, might be lagging behind. Considering all that, a more concise switch statement might be a better choice here.

Reply:
Agreed.